traitlets/traitlets.py

```diff
@@ -423,7 +428,9 @@ def __get__(self, obj, cls=None):
             except KeyError:
                 # Check for a dynamic initializer.
                 if self.name in obj._trait_dyn_inits:
-                    method = getattr(obj, obj._trait_dyn_inits[self.name])
+                    method = obj._trait_dyn_inits[self.name]
+                    if isinstance(method, str):
```

Comment:
Switching behavior based on `if isinstance` statements is not very object-oriented IMO.